### PR TITLE
Fix typed dimension display

### PR DESCRIPTION
**Pull request by gislenius**

### CODE REVIEW

**Comment by gislenius** [comment]
Sample to try:

http://rendering.secxbrl.info/#?url=http:%2F%2Fedinet-dev.28.io:80%2Fv1%2F_queries%2Fpublic%2Fapi%2Fspreadsheet-for-component.jq%3F_method%3DPOST%26token%3Dfoobar%26aid%3DS1002Q54%26format%3Dhtml%26role%3Dhttp%253A%252F%252Fdisclosure.edinet-fsa.go.jp%252Frole%252Fjppfs%252Frol_YearToQuarterEndConsolidatedStatementOfComprehensiveIncome%26profile-name%3Dgeneric%26eliminate%3Dtrue

File: queries/modules/io/28/modules/xbrl/profiles/sec/networks.jq [hunk 1]
```diff
@@ -439,20 +439,30 @@ declare function sec-networks:standard-definition-models-for-components($compone
...
+
     let $x-breakdowns as object* := (
         components:standard-period-breakdown()[not (($auto-slice-dimensions, $user-slice-dimensions) = "xbrl:Period")],
         for $d as string in $column-dimensions
         let $metadata as object? := ($component.Concepts[])[$$.Name eq $d]
+        let $label as string? :=
+          concepts:labels(
+            $d,
+            $concepts:VERBOSE_LABEL_ROLE,
+            ($options.Language, "en")[1],
```
Comment (dknochen) [inline queries/modules/io/28/modules/xbrl/profiles/sec/networks.jq]: in resolution.jq we use "en-US" as default. Is this still correct? Could we create a variable for the default language? Maybe, we have this already?

File: queries/modules/io/28/modules/xbrl/resolution.jq [hunk 2]
```diff
@@ -166,7 +166,7 @@ declare %private function resolution:convert-rule-node(
     let $labels := (
         $definition-node.Labels[],
         resolution:labels(
-            $main-members,
+            $main-members[$$ instance of string],
```
Comment (dknochen) [inline queries/modules/io/28/modules/xbrl/resolution.jq]: can we move the filtering to the assignment and make main-members typed?

**Comment by dknochen** [comment]
2 simple comments! thanks :)

File: queries/modules/io/28/modules/xbrl/profiles/sec/networks.jq (hunk 1 above)
Comment (gislenius) [inline queries/modules/io/28/modules/xbrl/profiles/sec/networks.jq]: That's done. I'm now using the component's default language. I refactored a bit as well.

File: queries/modules/io/28/modules/xbrl/facts.jq [hunk 3]
```diff
@@ -1372,7 +1372,7 @@ declare %private function facts:facts-for-direct(
           return
             (: add default dimension members if they are omitted
                remove dimensions that are not in the hypercube :)
-            copy $populated := $fact
+            copy $populated := trace($fact, "fact")
```
Comment (dknochen) [inline queries/modules/io/28/modules/xbrl/facts.jq]: trace

File: queries/modules/io/28/modules/xbrl/layout.jq [hunk 4]
```diff
@@ -479,16 +478,16 @@ declare function layout:layout(
...
         then ()
-        else ($options.Hypercube, hypercubes:dimensionless-hypercube())[1]
+        else ($options.Hypercube, hypercubes:dimensionless-hypercube())[1], "original")
     let $hypercube := layout:build-hypercube(
         $structural-model,
         $original-hypercube
     )
     let $facts := facts:facts-for({|
-        {Hypercube: $hypercube},
+        {Hypercube: trace($hypercube, "hypercube")},
```
Comment (dknochen) [inline queries/modules/io/28/modules/xbrl/layout.jq]: trace

File: queries/modules/io/28/modules/xbrl/layout.jq [hunk 5]
```diff
@@ -479,16 +478,16 @@ declare function layout:layout(
 {
     let $elimination as boolean := ($options.Eliminate, false)[1]
     let $threshold as double := ($options.EliminationThreshold, 0)[1]
-    let $original-hypercube :=
+    let $original-hypercube := trace(
```
Comment (dknochen) [inline queries/modules/io/28/modules/xbrl/layout.jq]: trace

**Pull request merged**